### PR TITLE
simplified regex (manually constructed)

### PR DESCRIPTION
**Pull request by panda2134**
This commit removed extraneous variables for empty substrings. Before this PR, every diagonal element can represent empty substrings, but now there's only one variable X_0_0 used for empty substrings.

### CODE REVIEW

File: benchmarks/regular-expressions/manually-constructed/regex8-aa.sl [hunk 1]
```diff
@@ -72,101 +72,101 @@
...
+                            (= X_0_2 (or (and A_0_0 B_0_2) (and A_0_1 B_1_2) (and A_0_2 B_0_0)))
+                            (= X_0_3 (or (and A_0_0 B_0_3) (and A_0_1 B_1_3) (and A_0_2 B_2_3) (and A_0_3 B_0_0)))
+                            (= X_0_4 (or (and A_0_0 B_0_4) (and A_0_1 B_1_4) (and A_0_2 B_2_4) (and A_0_3 B_3_4) (and A_0_4 B_0_0)))
+                            (= X_0_5 (or (and A_0_0 B_0_5) (and A_0_1 B_1_5) (and A_0_2 B_2_5) (and A_0_3 B_3_5) (and A_0_4 B_4_5) (and A_0_5 B_0_0)))
+                            (= X_0_6 (or (and A_0_0 B_0_6) (and A_0_1 B_1_6) (and A_0_2 B_2_6) (and A_0_3 B_3_6) (and A_0_4 B_4_6) (and A_0_5 B_5_6) (and A_0_6 B_0_0)))
+                            (= X_0_7 (or (and A_0_0 B_0_7) (and A_0_1 B_1_7) (and A_0_2 B_2_7) (and A_0_3 B_3_7) (and A_0_4 B_4_7) (and A_0_5 B_5_7) (and A_0_6 B_6_7) (and A_0_7 B_0_0)))
+                            (= X_0_8 (or (and A_0_0 B_0_8) (and A_0_1 B_1_8) (and A_0_2 B_2_8) (and A_0_3 B_3_8) (and A_0_4 B_4_8) (and A_0_5 B_5_8) (and A_0_6 B_6_8) (and A_0_7 B_7_8) (and A_0_8 B_0_0)))
+                            (= X_0_0 (and A_0_0 B_0_0))
```
Comment (kjcjohnson) [inline benchmarks/regular-expressions/manually-constructed/regex8-aa.sl]: Duplicate term `(= X_0_0 (and A_0_0 B_0_0))` + 7 more in this CHC. Others seem fine.

**Comment by kjcjohnson** [comment]
Everything except `regex8-aa.sl` looks fine. The `$concat` CHC in that one has some duplicate terms that my solver doesn't like.

File: benchmarks/regular-expressions/manually-constructed/regex8-aa.sl (hunk 1 above)
Comment (panda2134) [reply benchmarks/regular-expressions/manually-constructed/regex8-aa.sl]: fixed

**Comment by panda2134** [comment]
> Everything except `regex8-aa.sl` looks fine. The `$concat` CHC in that one has some duplicate terms that my solver doesn't like.

Just fixed.

**Pull request merged**